### PR TITLE
Added hw51 solution

### PR DESCRIPTION
**Pull request by IvanKrivoruchkoHub**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+    private Engine getEngine() {
+        return engine;
+    }
+
+    private void setEngine(Engine engine) {
+        this.engine = engine;
+    }
+
+    public Car clone() throws CloneNotSupportedException {
+        Car clonedCar = (Car) super.clone();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]: try to solve without super.clone()

File: src/main/java/core/basesyntax/Car.java [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
 
+    public Car(String color, String name, int maxSpeed, int price, Engine engine) {
+        this.color = color;
+        this.name = name;
+        this.maxSpeed = maxSpeed;
+        this.price = price;
+        this.engine = engine;
+    }
+
+    public Car clone() throws CloneNotSupportedException {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]: we should override this method, not create new one

File: src/main/java/core/basesyntax/Engine.java [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
+    public Engine(int powerful, int price, int weight,
+                  int numberOfCylinders, String type) {
+        this.powerful = powerful;
+        this.price = price;
+        this.weight = weight;
+        this.numberOfCylinders = numberOfCylinders;
+        this.type = type;
+    }
+
+    public Engine clone() throws CloneNotSupportedException {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Engine.java]: same

Comment (IvanKrivoruchkoHub) [reply src/main/java/core/basesyntax/Engine.java]: I don't understand. This method is overridden. When we override a method we can extend its access specifier and narrow the type that return method.

Comment (Tarabarnik) [reply src/main/java/core/basesyntax/Engine.java]: > I don't understand. This method is overridden. When we override a method we can extend its access specifier and narrow the type that return method.
To create correct override of method we should add `@Override` annotation above it.
P.S. для швидшої відповіді, пиши в Slack

**Pull request closed without merging**